src/common/QuirksManager.cpp

```diff
@@ -103,10 +103,24 @@ class SamsungDeviceQuirks : public  QuirksManager::DeviceQuirks {
         // This detects b/159066712 , S20 LSI has corrupt low latency audio recording
         // and turns off MMAP.
         // See also https://github.com/google/oboe/issues/892
-        bool mRecordingCorrupted = isInput
+        bool isRecordingCorrupted = isInput
             && isExynos990
             && mBuildChangelist < 19350896;
-        return !mRecordingCorrupted;
+
+        // Certain S9+ builds record silence when using MMAP and not using the VoiceCommunication
+        // preset.
+        // See https://github.com/google/oboe/issues/1110
+        bool wouldRecordSilence = isInput
+            && isExynos9810
```

Comment:
Do we need this for the other Samsung devices as well?

Reply:
Maybe, seems like this issue on S10 is similar: https://github.com/google/oboe/issues/1320. 

If so, we can always update this quirk to include that device in a separate PR.